solidity/test/LiquidityProtection.js

```diff
@@ -2223,106 +2223,54 @@ describe('LiquidityProtection', () => {
-                                ),
-                                'ERR_REVERT'
-                            );
-                        });
-
-                        it('should revert when calling an invalid method', async () => {
-                            await expectRevert(
-                                liquidityProtection.transferPositionAndCall(
-                                    protectionId,
-                                    newOwner,
-                                    testCall.address,
-                                    liquidityProtection.contract.methods.store().encodeABI(),
-                                    {
-                                        from: recipient
-                                    }
-                                ),
-                                'ERR_CALL_FAILED'
+                                'ERR_INVALID_ADDRESS'
                             );
                         });
 
-                        it('should support state changing function', async () => {
-                            expect(await testCall.num.call()).to.be.bignumber.equal(new BN(0));
-                            expect(await testCall.str.call()).to.be.eql('');
+                        it('should notify on transfer', async () => {
+                            const transferEvent = await callback.transferEvent.call();
+                            expect(transferEvent[0]).to.be.bignumber.equal(new BN(0));
+                            expect(transferEvent[1]).to.eql(ZERO_ADDRESS);
+                            expect(transferEvent[2]).to.be.null();
 
-                            const newNum = new BN(12345);
-                            const newStr = 'Hello World!';
+                            const data = '0x1234';
 
                             await verifyTransfer(async () =>
-                                liquidityProtection.transferPositionAndCall(
+                                liquidityProtection.transferPositionAndNotify(
                                     protectionId,
                                     newOwner,
-                                    testCall.address,
-                                    testCall.contract.methods.set(newNum, newStr).encodeABI(),
+                                    callback.address,
+                                    data,
                                     {
                                         from: recipient
                                     }
                                 )
                             );
 
-                            expect(await testCall.num.call()).to.be.bignumber.equal(newNum);
-                            expect(await testCall.str.call()).to.be.eql(newStr);
+                            const protectionIds2 = await liquidityProtectionStore.protectedLiquidityIds(newOwner);
+                            expect(protectionIds2.length).to.eql(1);
+
+                            const transferEvent2 = await callback.transferEvent.call();
```

Comment:
We might wanna get rid of `.call()` conclusively at some point.
It's a bit off from web3.js syntax, but that's already the case here with `.method` missing, so why not make the code cleaner?

Reply:
`.cal()` isn't required in V3 already (due to hardhat)

Reply:
`.cal()` isn't required in any of the previous versions either (due to truffle)., hence this comment.